src/internal/operators/windowTime.ts

```diff
@@ -97,188 +101,100 @@ export function windowTime<T>(windowTimeSpan: number,
-    this.destination.complete();
-  }
-
-  public openWindow(): CountedSubject<T> {
-    const window = new CountedSubject<T>();
-    this.windows.push(window);
-    const destination = this.destination;
-    destination.next(window);
-    return window;
-  }
-
-  public closeWindow(window: CountedSubject<T>): void {
-    const index = this.windows.indexOf(window);
-    // All closed windows should have been removed,
-    // we don't need to call complete unless they're found.
-    if (index >= 0) {
-      window.complete();
-      this.windows.splice(index, 1);
-    }
-  }
-}
-
-function dispatchWindowTimeSpanOnly<T>(this: SchedulerAction<TimeSpanOnlyState<T>>, state: TimeSpanOnlyState<T>): void {
-  const { subscriber, windowTimeSpan, window } = state;
-  if (window) {
-    subscriber.closeWindow(window);
-  }
-  state.window = subscriber.openWindow();
-  this.schedule(state, windowTimeSpan);
-}
-
-function dispatchWindowCreation<T>(this: SchedulerAction<CreationState<T>>, state: CreationState<T>): void {
-  const { windowTimeSpan, subscriber, scheduler, windowCreationInterval } = state;
-  const window = subscriber.openWindow();
-  const action = this;
-  let context: CloseWindowContext<T> = { action, subscription: null! };
-  const timeSpanState: CloseState<T> = { subscriber, window, context };
-  context.subscription = scheduler.schedule<CloseState<T>>(dispatchWindowClose as any, windowTimeSpan, timeSpanState);
-  action.add(context.subscription);
-  action.schedule(state, windowCreationInterval);
+      };
+
+      windowCreationInterval !== null && windowCreationInterval >= 0
+        ? // The user passed both a windowTimeSpan (required), and a creation interval
+          // That means we need to start new window on the interval, and those windows need
+          // to wait the required time span before completing.
+          subscriber.add(
+            scheduler.schedule(function () {
+              startWindow();
+              !this.closed && subscriber.add(this.schedule(null, windowCreationInterval));
+            }, windowCreationInterval)
+          )
+        : (restartOnClose = true);
+
+      startWindow();
+
+      /**
+       * We need to loop over a copy of the window records several times in this operator.
+       * This is to save bytes over the wire more than anything.
+       * The reason we copy the array is that reentrant code could mutate the array while
+       * we are iterating over it.
+       */
+      const loop = (cb: (record: WindowRecord<T>) => void) => windowRecords!.slice().forEach(cb);
+
+      /**
+       * Used to notify all of the windows and the subscriber in the same way
+       * in the error and complete handlers.
+       */
+      const terminate = (cb: (consumer: Observer<any>) => void) => {
+        loop(({ window }) => cb(window));
+        cb(subscriber);
+        windowRecords = null!;
+        subscriber.unsubscribe();
+      };
+
+      const windowTimeSubscriber = new OperatorSubscriber(
```

Comment:
nitpick: as with `windowCount`, this doesn't really need to be a variable.

Reply:
Done... Also added a bit to ensure the buffer here is released.